src/compiler/parser.ts

```diff
@@ -3160,6 +3160,11 @@ module ts {
                 return undefined;
             }
 
+            // Must be no line terminator before token `=>`.
+            if (scanner.hasPrecedingLineBreak()) {
+                return undefined;

```

Comment:
Wow, that was quick! I started a bit while I was in the office, but if you'd like to take this task on I wouldn't mind.

I actually had the idea of continuing to parse arrow functions exactly as before, but to report the error as part of our grammatical checking in `checker.ts`. Since `=> blah(blah, blahblah)` is not a valid expression in any context anyway, this should be acceptable, and it's ideal for language service scenarios.

However, the problem I ran into when I tried to fix this was that it's a real pain to work on the token-level in the checker.  So I wonder if the benefit of the idea is actually worth the cost. When I discussed with @CyrusNajmabadi, he seemed to favor the idea, but I'd like him to weigh in.